File: Winter2020/lab-solutions/Lab7/RoomPersistenceLabSkeleton/app/src/main/java/mobiledev/unb/ca/roompersistencelab/ItemsAdapter.java
package mobiledev.unb.ca.roompersistencelab;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import java.util.List;

public class ItemsAdapter extends ArrayAdapter<Item> {
    public ItemsAdapter(Context context, List<Item> items) {
        super(context, 0, items);
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        // Get the data item for this position
        Item item = getItem(position);
        // Check if an existing view is being reused, otherwise inflate the view
        if (convertView == null) {
            convertView = LayoutInflater.from(getContext()).inflate(R.layout.list_layout, parent, false);
        }

        // Lookup view for data population
        TextView tvName = convertView.findViewById(R.id.item_textview);
        TextView tvNum = convertView.findViewById(R.id.num_textview);

        // Populate the data into the template view using the data object
        tvName.setText(item.getName());
        tvNum.setText(String.valueOf(item.getNum()));

        // Return the completed view to render on screen
        return convertView;
    }
}

File: Winter2022/examples/kotlin/Lecture 7/RoomPersistenceLibraryDemo/app/src/main/java/mobiledev/unb/ca/roompersistencetest/ui/ItemsAdapter.kt
package mobiledev.unb.ca.roompersistencetest.ui

import android.content.Context
import android.widget.ArrayAdapter
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import mobiledev.unb.ca.roompersistencetest.R
import android.widget.TextView
import mobiledev.unb.ca.roompersistencetest.entity.Item

class ItemsAdapter(context: Context, items: List<Item>) : ArrayAdapter<Item>(
    context, 0, items) {
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        // Get the data item for this position
        var view = convertView
        val item = getItem(position)

        // Check if an existing view is being reused, otherwise inflate the view
        if (view == null) {
            view = LayoutInflater.from(context).inflate(R.layout.list_layout, parent, false)
        }

        // Lookup view for data population
        val tvName = view!!.findViewById<TextView>(R.id.item_textview)
        val tvNum = view.findViewById<TextView>(R.id.num_textview)

        // Populate the data into the template view using the data object
        tvName.text = item!!.name
        tvNum.text = item.num.toString()

        // Return the completed view to render on screen
        return view
    }
}
